src/utils/front-matter.js

```diff
@@ -5,38 +5,56 @@ const {
   builders: { hardline, concat, markAsRoot },
 } = require("../document");
 
+const YAML_DELIMITER = "---";
+// In some markdown processors such as pandoc,
+// "..." can be used as the end delimiter for YAML front-matter.
+const YAML_END_DELIMITER = "...";
 const DELIMITER_MAP = {
-  "---": "yaml",
+  [YAML_DELIMITER]: "yaml",
   "+++": "toml",
```

Comment:
How about save delimiters like this

```js
['---', '---'],
['---', '...'],
['+++', '+++']
```

Reply:
Where were you thinking of using the array?

Reply:
I was thinking we should separate the RegExp. Like

```js
^---(LANG)(CONTENT)--- |
^---(CONTENT)... |
^+++(CONTENT)+++ 
```

Reply:
That would change the behavior I think, since right now I can have (probably unintentionally):

```
+++toml
x = 5
+++
```
parse as TOML

Reply:
@fisker do you think it's ok to leave it as is? There doesn't seem to be much of a spec for the behavior here so I'm hesitant to change anything besides `...` in this PR